warehouse/query-core/src/main/java/datawave/query/jexl/visitors/ExecutableDeterminationVisitor.java

```diff
@@ -594,390 +405,1040 @@ public Object visit(ASTLTNode node, Object data) {
+                throw new RuntimeException("got exception when using MetadataHelper to get index only fields", e);
+            }
+        }
+        
+        return JexlASTHelper.getIdentifiers(node).stream().map(JexlASTHelper::deconstructIdentifier).anyMatch(this.indexOnlyFields::contains);
+    }
+    
+    /**
+     * Returns whether or not the specified nodes has any non-event identifiers.
+     *
+     * @param node
+     *            the node
+     * @return true if the node has any any non-event identifiers, or false otherwise.
+     * @throws RuntimeException
+     *             if the non-event fields need to be retrieved from the metadata helper and an exception occurs
+     */
+    private boolean isNonEvent(JexlNode node) {
+        // Initialize the non-event fields if necessary.
+        if (this.nonEventFields == null) {
+            try {
+                this.nonEventFields = helper.getNonEventFields(config.getDatatypeFilter());
+            } catch (TableNotFoundException e) {
+                log.error("Could not determine whether field is non event", e);
+                throw new RuntimeException("got exception when using MetadataHelper to get non event fields", e);
+            }
+        }
+        
+        return JexlASTHelper.getIdentifiers(node).stream().map(JexlASTHelper::deconstructIdentifier).filter(ident -> !ident.equals(Constants.NO_FIELD))
+                        .anyMatch(this.nonEventFields::contains);
+    }
+    
+    /**
+     * Returns whether or not if the specified node is part of an AND for a bounded range, e.g. {@literal (A < 12 && A > 4)}.
+     *
+     * @param node
+     *            the node
+     * @return true if the node is within a bounded range or false otherwise.
+     */
+    private boolean isWithinBoundedRange(JexlNode node) {
+        if (node.jjtGetParent() instanceof ASTAndNode) {
+            List<JexlNode> otherNodes = new ArrayList<>();
+            Map<LiteralRange<?>,List<JexlNode>> ranges = JexlASTHelper.getBoundedRangesIndexAgnostic(node.jjtGetParent(), otherNodes, false);
+            return ranges.size() == 1 && otherNodes.isEmpty();
+        }
+        return false;
+    }
+    
+    /**
+     * Populates the specified set with the states returned by each child of the specified node when the child accepts this visitor.
+     *
+     * @param node
+     *            the node
+     * @param data
+     *            the data
+     * @param states
+     *            the set of states
+     */
+    private void readStatesFromChildren(final JexlNode node, final Object data, final Set<STATE> states) {
+        final int numChildren = node.jjtGetNumChildren();
+        for (int i = 0; i < numChildren; i++) {
+            states.add((STATE) (node.jjtGetChild(i).jjtAccept(this, data)));
+            // Break early if all possibly unique states have already been added.
+            if (states.size() == STATE.size) {
```

Comment:
this is a good idea, but we need a flag to bypass this for debugging

Reply:
Done.